fundamentals/middleware/rate-limit/WebRateLimitAuth/Program.cs

```diff
@@ -0,0 +1,447 @@
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => 
+                  options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+
+builder.Services.AddRazorPages();
+builder.Services.AddControllersWithViews();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseMigrationsEndPoint();
+}
+else
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+app.UseAuthentication();
+app.UseAuthorization();
+
+// <snippet_adm2>
+var getPolicyName = "get";
+var postPolicyName = "post";
+var myOptions = new MyRateLimitOptions();
+app.Configuration.GetSection(MyRateLimitOptions.MyRateLimit).Bind(myOptions);
+
+app.UseRateLimiter(new RateLimiterOptions()
+    .AddConcurrencyLimiter(policyName: getPolicyName,
+          new ConcurrencyLimiterOptions(permitLimit: myOptions.permitLimit,
+          queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+          queueLimit: myOptions.queueLimit))
+    .AddPolicy(policyName: postPolicyName, partitioner: httpContext =>
+    {
+        string userName = httpContext?.User?.Identity?.Name ?? string.Empty;
+
+        if (!StringValues.IsNullOrEmpty(userName))
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter(userName, key =>
+                new TokenBucketRateLimiterOptions(tokenLimit: myOptions.tokenLimit2,
+                    queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+                    queueLimit: myOptions.queueLimit,
+                    replenishmentPeriod: TimeSpan.FromSeconds(myOptions.replenishmentPeriod),
+                    tokensPerPeriod: myOptions.tokensPerPeriod,
+                    autoReplenishment: myOptions.autoReplenishment));
+        }
+        else
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter("Anon", key =>
+                new TokenBucketRateLimiterOptions(tokenLimit: myOptions.tokenLimit,
+                    queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+                    queueLimit: myOptions.queueLimit,
+                    replenishmentPeriod: TimeSpan.FromSeconds(myOptions.replenishmentPeriod),
+                    tokensPerPeriod: myOptions.tokensPerPeriod,
+                    autoReplenishment: true));
+        }
+    }));
+// </snippet_adm2>
+
+static string GetUserEndPointMethod(HttpContext context) =>
+    $"Hello {context.User?.Identity?.Name ?? "Anonymous"} " +
+    $"Endpoint:{context.Request.Path} Method: {context.Request.Method}";
+
+app.MapGet("/test", (HttpContext context) => $"{GetUserEndPointMethod(context)}")
+                                        .RequireRateLimiting(getPolicyName);
+
+app.MapRazorPages().RequireRateLimiting(getPolicyName)
+                   .RequireRateLimiting(postPolicyName);
+
+app.MapDefaultControllerRoute();
+
+app.Run();
+// </snippet_adm>
+#elif JWT
```

Comment:
> > It's better to also include JWT Auth handler, so it is clear in the sample that where does the "access_token" come form. `builder.Services.AddAuthentication().AddJwt(...);`

@Kahbazi I made a new sample for this and removed `GetTokenValue("access_token")` from the other sample. This one is easy to test with [dotnet user-jwts](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/jwt-authn?view=aspnetcore-7.0&tabs=windows)